zerver/webhooks/slack/view.py

```diff
@@ -38,5 +38,12 @@ def api_slack_webhook(
     content = ZULIP_MESSAGE_TEMPLATE.format(message_sender=user_name, text=text)
     client = RequestNotes.get_notes(request).client
     assert client is not None
-    check_send_stream_message(user_profile, client, stream, topic, content)
+    check_send_webhook_message(
+        request,
+        user_profile,
+        topic,
+        content,
+        stream=stream,
```

Comment:
Wouldn't this, as implemented, always ignore any stream that was in the generated webhook URL? I think we'd only want to set the stream if the topic wasn't specified in the URL and the `channels_map_to_topics` was false (aka 0).

Reply:
It looks like your updated changes still don't account for a stream ID or name in the URL that's submitted. I'll note that if a topic is in the URL, then it's expected that there is also a stream identified in the URL.

Reply:
@laurynmm 
I included the (stream=stream) parameter to handle the case where the topic wasn't specified in the URL and the channels_map_to_topics was false.
I have tested in the integrations developer panel, the code always picks up the stream query parameter mentioned in the URL. 
Am I missing something?

Reply:
> It looks like your updated changes still don't account for a stream ID or name in the URL that's submitted. I'll note that if a topic is in the URL, then it's expected that there is also a stream identified in the URL.

I didn't understand this bit, what exactly do I have to do?

Reply:
> Wouldn't this, as implemented, always ignore any stream that was in the generated webhook URL? I think we'd only want to set the stream if the topic wasn't specified in the URL and the `channels_map_to_topics` was false (aka 0).

In my new code, would you consider this query as resolved?

Reply:
@laurynmm ,what shall I do?